### PR TITLE
FSUI: Add Recording Indicator and Time

### LINKED ISSUE #10307: [Feature Request]: Recording Time on OSD
### Description

I would like for the OSD to be able to display how long the current recording session has gone for, both in terms of real-time and the current length of the video file.

### Reason

I use the internal recorder for creating post-commentary YouTube playthroughs. I try to keep my video lengths more or less consistent, but this can be hard when there's no indication of how long the recording has gone, which often results in more editing of the footage being necessary. I can set a manual timer when I start and try to remember to turn it off when I end, but this only tracks the amount of real time, which can be very different from the final length of the video if the game speeds up or slows down for any reason, which is common (game is more likely to slow down during recording and I often speed up the game during cutscenes).

It would also be a clear indicator that recording is happening, which can be a bit difficult to determine right now.

### Examples

The Xbox Game Bar built in to Windows displays a small overlay when recording. As it is not an internal emulator recording feature, it only makes sense for it to display the current amount of real time.
Game Bar

### PR DESCRIPTION
**Pull request by kamfretoz**
### Description of Changes

This PR adds an OSD indicator and elapsed timer for the Video Capture.

Closes #10307.

currently only real time tracking are implemented.

Screenshot_20240113_133231

### Rationale behind Changes

Quality of Life improvements.

### Suggested Testing Steps

Test possible edge cases like pausing while recording, etc.

### CODE REVIEW

**Comment by kamfretoz** [comment]
Added File Name and Total Frames indicator.

Screenshot_20240113_133231

**Review by Mrlinkwii — APPROVED** [review]
works good , would have the suggestion to remove the frame counting ,

File: pcsx2/GS/GSCapture.cpp [hunk 1]
```diff
@@ -1339,6 +1345,11 @@ bool GSCapture::IsCapturingAudio()
 	return (s_audio_stream != nullptr);
 }
 
+TinyString GSCapture::GetElapsedTime() {
```
[the remark this answers is not in the document]
Comment (kamfretoz) [reply pcsx2/GS/GSCapture.cpp]: Old habit die hard, Fixed.

**Review by Mrlinkwii — APPROVED** [review]

**Comment by kamfretoz** [comment]
Made the text more compact.
image

File: pcsx2/GS/GSCapture.cpp [hunk 2]
```diff
@@ -122,6 +123,8 @@ extern "C" {
 
 namespace GSCapture
 {
+	Common::Timer s_capture_start_time;
```
Comment (stenzek) [inline pcsx2/GS/GSCapture.cpp:126]: Can you move this down to where the rest of the variables are? (line 163 or so). Constants are at the beginning.

Also, should be static.

File: pcsx2/ImGui/ImGuiOverlays.cpp [hunk 3]
```diff
@@ -652,6 +653,40 @@ void ImGuiManager::DrawInputRecordingOverlay(float& position_y)
...
+	const float scale = ImGuiManager::GetGlobalScale();
+	const float shadow_offset = std::ceil(1.0f * scale);
+	const float margin = std::ceil(10.0f * scale);
+	const float spacing = std::ceil(5.0f * scale);
+	position_y += margin;
+
+	ImFont* const standard_font = ImGuiManager::GetStandardFont();
+
+	ImDrawList* dl = ImGui::GetBackgroundDrawList();
+	std::string text;
```
Comment (stenzek) [inline pcsx2/ImGui/ImGuiOverlays.cpp:667]: This variable is not used.

**Comment by SuperfieldCU** [comment]
> works good , would have the suggestion to remove the frame counting ,

This is actually something I requested in the original feature request (although having it presented as a time code would be preferred over frame count) - knowing how long the final recording will be is more valuable information for my use case than the amount of real time that the recording has gone for (since the emulation tends to slow down more often when recording and I will often speed up the emulation during cutscenes).

**Comment by kamfretoz** [comment]
> This is actually something I requested in the original feature request (although having it presented as a time code would be preferred over frame count) - knowing how long the final recording will be is more valuable information for my use case than the amount of real time that the recording has gone for (since the emulation tends to slow down more often when recording and I will often speed up the emulation during cutscenes).

Unfortunately, tracking the final length of the video requires more thorough changes to the video capture code and I'm unable to do so at the moment.

Maybe that's something to do for the future.
